ssl/ssl_test.cc

```diff
@@ -5014,6 +5023,113 @@ TEST(SSLTest, EmptyCipherList) {
   EXPECT_EQ(0u, sk_SSL_CIPHER_num(SSL_CTX_get_ciphers(ctx.get())));
 }
 
+struct MultiTransferReadWriteTestParams {
+  std::string suite;
+  bool tls13;
+  bssl::UniquePtr<X509> (*certificate)();
+  bssl::UniquePtr<EVP_PKEY> (*key)();
+};
+
+static const MultiTransferReadWriteTestParams kMultiTransferReadWriteTests[] = {
```

Comment:
We can always refactor it in the future if needed, but I opted for the simplest solution given that there are no on-going drafts (that I have seen) to add a new TLS version (for example 1.4), which would have additional work besides implementing TLS transfer. Additionally cipher suites aren't being added at an alarming rate, given that since TLS 1.3 being added we haven't extended support past the initial three ciphers we have offered.